common/djangoapps/student/models.py

```diff
@@ -1318,6 +1323,19 @@ def is_enrollment_closed(cls, user, course):
         from openedx.core.djangoapps.enrollments.permissions import ENROLL_IN_COURSE
         return not user.has_perm(ENROLL_IN_COURSE, course)
 
+    def has_proctoring_requirements(self, proctored_exams_enabled):
+        """
+        Returns a boolean value regarding whether course mode requires a proctored exam. Returns False if special
+        exams aren't enabled, proctored exams aren't enabled for the course, or if the course mode is audit.
+        """
+        if not settings.FEATURES.get('ENABLE_SPECIAL_EXAMS'):
+            return False
+        if not proctored_exams_enabled:
+            return False
+        if self.mode not in CourseMode.CERTIFICATE_RELEVANT_MODES:
```

Comment:
I'm wondering if there is a way to have a single place for this logic. This is also determined here https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/rules.py#L34

Reply:
Thanks, I didn't realize that was there! I think it makes sense for it to be in the same place, I think I just have to modify the logic slightly to use a course rather than an exam.